### PR TITLE
creates alarms

### PR DESCRIPTION
**Pull request by humbertouchiyama**

### CODE REVIEW

File: app/src/main/java/com/humberto/tasky/core/alarm/data/AgendaAlarmScheduler.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        val now = System.currentTimeMillis()
+        if (alarmItem.triggerAt < now) return
+
+        val intent = Intent(context, AlarmReceiver::class.java).apply {
+            putExtra(AlarmReceiver.ALARM_ID, alarmItem.id)
+            putExtra(AlarmReceiver.ALARM_TITLE, alarmItem.title)
+            putExtra(AlarmReceiver.ALARM_DESCRIPTION, alarmItem.description)
+            putExtra(AlarmReceiver.ITEM_TYPE, alarmItem.itemType.name)
+            putExtra(AlarmReceiver.ITEM_DATE, alarmItem.itemDate)
+        }
```
[the remark this answers is not in the document]
Comment (humbertouchiyama) [reply app/src/main/java/com/humberto/tasky/core/alarm/data/AgendaAlarmScheduler.kt:29]: But in this case I'm only using `val triggerAt: Long` on `alarmManager.setExactAndAllowWhileIdle(long triggerAtMillis)` here on the scheduler, isn't better to just have it Long?

[the remark this answers is not in the document]
Comment (humbertouchiyama) [reply app/src/main/java/com/humberto/tasky/core/alarm/data/AgendaAlarmScheduler.kt:29]: Do you not pass the itemType here when creating the Intent URI to the deep link?

[the remark this answers is not in the document]
Comment (Kai-Cilliers) [reply app/src/main/java/com/humberto/tasky/core/alarm/data/AgendaAlarmScheduler.kt:29]: I differentiated between the agenda items using a prefix to the ID parameter of each

When creating an item (I created agenda items and saved them to the DB after user selects the agenda item via the FAB on Agenda screen)
```
val eventId = AgendaItem.PREFIX_EVENT_ID + UUID.randomUUID()
```

Then I would check them:
```
when {
  itemId.contains(AgendaItem.PREFIX_TASK_ID) -> /* Do something */
  itemId.contains(AgendaItem.PREFIX_REMINDER_ID) -> /* Do something */
  itemId.contains(AgendaItem.PREFIX_EVENT_ID) -> /* Do something */
}
```

[the remark this answers is not in the document]
Comment (Kai-Cilliers) [reply app/src/main/java/com/humberto/tasky/core/alarm/data/AgendaAlarmScheduler.kt:29]: Sure, you can. However, it is still recommended to have two models: one parcelable and one (your domain) which is not. You should not be leaking implementation details (like parcelability) to your domain.

**Pull request merged**